significant_strikes.py

```diff
@@ -0,0 +1,500 @@
+
+        if not all_prev_fights.empty:
+            # Filter fights involving the specified fighter
+            fighter_a_id_vals = all_prev_fights.fighter_a_id.values
+            fighter_b_id_vals = all_prev_fights.fighter_b_id.values
+
+            prev_fights = all_prev_fights[((fighter_a_id_vals == fighter_id) | (fighter_b_id_vals == fighter_id))]
+            num_fights = len(prev_fights)
+
+            # Initialize empty DataFrames for the scenarios where there are fewer than 3 or 5 fights
+            last_3_fights = pd.DataFrame()
+            last_5_fights = pd.DataFrame()
+
+            # Check if the fighter has at least 3 fights
+            if num_fights >= 3:
+                last_3_fights = prev_fights.tail(3)
+                
+            else:
+                # If fewer than 3 fights, return all available fights
+                last_3_fights = prev_fights
+
+            # Check if the fighter has at least 5 fights
+            if num_fights >= 5:
+                last_5_fights = prev_fights.tail(5)
+                
+            else:
+                # If fewer than 5 fights, return all available fights
+                last_5_fights = prev_fights
+                
+            return last_3_fights, last_5_fights, prev_fights
+
+        return pd.DataFrame(), pd.DataFrame(), pd.DataFrame()
+        
+
+    def get_raw_significant_strikes_stats(self, df, round):
+        """
+        Extracts raw significant strike stats from the dataframe for a specific round.
+
+        Args:
+            df (pd.DataFrame): The dataframe containing fight records.
+            round (str): The round for which the statistics are required.
+
+        Returns:
+            pd.DataFrame: A dataframe with raw stats for the specified round for fighters a and b.
+        """
+
+        # Define column names based on the round
+        if round == "overall":
+            # Use overall stats columns
+            sig_str_cols = [
+                'fighter_a_total_sig_str_landed',
+                'fighter_a_total_sig_str_attempted',
+                'fighter_b_total_sig_str_landed',
+                'fighter_b_total_sig_str_attempted'
+            ]
+        else:
+            # Use specific round stats columns
+            sig_str_cols = [
+                f'fighter_a_round_{round[1]}_sig_str_landed',
+                f'fighter_a_round_{round[1]}_sig_str_attempted',
+                f'fighter_b_round_{round[1]}_sig_str_landed',
+                f'fighter_b_round_{round[1]}_sig_str_attempted'
+            ]
+
+        # Initialize an empty DataFrame with the specified columns
+        stats_df = pd.DataFrame(columns=[
+            'total_fight_time', 'fighter_a_id', 'fighter_b_id',
+            'fighter_a_sig_str_landed', 'fighter_a_sig_str_attempted',
+            'fighter_b_sig_str_landed', 'fighter_b_sig_str_attempted'
+        ])
+
+        # Iterate through each fight in the DataFrame
+        for index, fight in df.iterrows():
```

Comment:
For this for loop I think you can make it faster and more concise:

`stats_df = pd.concat([fight.apply(lambda row: self.get_round_time(row['outcome_round'], row['outcome_time'], round), fight['fighter_a_id], fight['fighter_b_id'], [fight[col] for col in sig_str_cols], ], keys=['total_fight_time', fighter_a_id', 'fighter_b_id', ...], axis=1)`

I'm not entirely sure of the syntax, but you can definitely vectorize this operation and concat an apply function with other vectorization techniques to make this much faster. Speed is definitely important when it comes to this feature set because it's a very large set.